source/includes/write/delete.scala

```diff
@@ -0,0 +1,57 @@
+import org.mongodb.scala._
+import org.mongodb.scala.bson.Document
+import org.mongodb.scala.model.Filters.{and, equal, regex}
+import org.mongodb.scala.model.{DeleteOptions}
+import org.mongodb.scala.result.{DeleteResult}
+
+object Delete {
+
+  def main(args: Array[String]): Unit = {
+    val mongoClient = MongoClient("<connection string URI>")
+
+    // start-db-coll
+    val database: MongoDatabase = mongoClient.getDatabase("sample_restaurants")
+    val collection: MongoCollection[Document] = database.getCollection("restaurants")
+    // end-db-coll
+
+    // start-delete-one
+    val filter = equal("name", "Happy Garden")
+    val observable: Observable[DeleteResult] = collection.deleteOne(filter)
+
+    observable.subscribe(new Observer[DeleteResult] {
+      override def onNext(result: DeleteResult): Unit = println(s"Deleted document count: ${result.getDeletedCount}")
+      override def onError(e: Throwable): Unit = println(s"Error: $e")
```

Comment:
```suggestion
      override def onNext(result: DeleteResult): Unit = println("Deleted document count: ${result.getDeletedCount}")
      override def onError(e: Throwable): Unit = println("Error: $e")
```
Q: Not yet familiar with Scala, is the 's' necessary here in the print statements?

Reply:
This is a way to do string interpolation: [link to documentation](https://docs.scala-lang.org/scala3/book/string-interpolation.html)

Reply:
Thanks, looks good!